File: backend/app/schemas/analytics.py
"""Pydantic schemas for analytics endpoints."""

import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class AnalyticsSnapshotResponse(BaseModel):
    id: uuid.UUID
    published_post_id: uuid.UUID
    impressions: int
    likes: int
    comments: int
    shares: int
    clicks: int
    engagement_rate: Decimal | None
    snapshot_at: datetime

    model_config = {"from_attributes": True}


class PostAnalytics(BaseModel):
    post_id: uuid.UUID
    platform: str
    content_text: str
    published_at: datetime | None
    platform_url: str | None
    impressions: int
    likes: int
    comments: int
    shares: int
    clicks: int
    engagement_rate: Decimal | None


class AnalyticsSummary(BaseModel):
    total_posts: int
    total_impressions: int
    total_likes: int
    total_comments: int
    total_shares: int
    avg_engagement_rate: float
    posts: list[PostAnalytics]


class DashboardStats(BaseModel):
    total_posts: int
    published_posts: int
    scheduled_posts: int
    draft_posts: int
    total_impressions: int
    total_likes: int
    avg_engagement_rate: float
    recent_posts: list[dict]
    upcoming_posts: list[dict]

File: frontend/src/api/analytics.ts
import api from "./client"

export interface PostAnalytics {
  post_id: string
  platform: string
  content_text: string
  published_at: string | null
  platform_url: string | null
  impressions: number
  likes: number
  comments: number
  shares: number
  clicks: number
  engagement_rate: number | null
}

export interface AnalyticsSummary {
  total_posts: number
  total_impressions: number
  total_likes: number
  total_comments: number
  total_shares: number
  avg_engagement_rate: number
  posts: PostAnalytics[]
}

export interface DashboardStats {
  total_posts: number
  published_posts: number
  scheduled_posts: number
  draft_posts: number
  total_impressions: number
  total_likes: number
  avg_engagement_rate: number
  recent_posts: Array<{
    id: string
    platform: string
    text: string
    status: string
    updated_at: string
  }>
  upcoming_posts: Array<{
    id: string
    platform: string
    text: string
    scheduled_for: string | null
  }>
}

export async function fetchAnalytics(params?: {
  days?: number
  platform?: string
}): Promise<AnalyticsSummary> {
  const { data } = await api.get("/analytics", { params })
  return data
}

export async function fetchDashboard(): Promise<DashboardStats> {
  const { data } = await api.get("/dashboard")
  return data
}
